src/main/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStep.java

```diff
@@ -48,6 +51,11 @@ public SSHAgentStep(final List<String> credentials) {
         this.ignoreMissing = false;
     }
 
+    @Override
+    public StepExecution start(StepContext context) throws Exception {
+        return new SSHAgentStepExecution(this, context);
+    }
+
     @Extension
     public static final class DescriptorImpl extends AbstractStepDescriptorImpl {
```

Comment:
This Guiceful class would need to be converted to nondeprecated `StepDescriptor`.